### DISCUSSION TITLE
How do I increase the font size of copilot chat?

### DISCUSSION THREAD

**Discussion by Iroegbu1** [community/community#55469] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Please, how can I increase the font size of the chat responses, it's currently too tiny at about 12px.

**Comment by RustingSword** [comment, marked as answer]
A quick workaround is to zoom in the whole UI of vs code, then adjust the font size of editor/terminal etc. accordingly. See https://stackoverflow.com/a/51464582

**Reply by Iroegbu1** [reply]
You're the real MVP.

**Reply by Akash1134** [reply]
Thank you for sharing this. ❤️

**Reply by NawNow1362** [reply]
HIFIDIBO

**Reply by derwaldgeist** [reply]
This does not work for WebStorm. :-/

**Reply by garv999** [reply]
🔹 GitHub Copilot Chat Font Size – How to Increase
GitHub Copilot uses your IDE’s default font size settings. It does not have a separate font size setting for the chat responses.

✅ Steps to Increase Font Size in VS Code:
Go to Settings:

Press Ctrl + , (Windows) or Cmd + , (Mac).

Search for Font Size:

Type font size in the search bar.

Adjust the Font Size:

Change the value of Editor: Font Size (e.g., from 12 to 16).

Zoom the Entire Interface:

Press Ctrl + + (Windows) or Cmd + + (Mac) to zoom in.

This increases the size of all UI elements, including Copilot Chat.

Adjust Terminal Font Size Separately (if needed):

Search for Terminal Integrated Font Size in settings to separately increase the terminal text size.

🔹 Important:
There is currently no separate font size setting specifically for GitHub Copilot Chat. It inherits your VS Code settings.

**Reply by PollySantos** [reply]
Para aumentar a fonte do chat do GitHub Copilot no VS Code, você pode simplesmente aumentar o zoom do editor usando Ctrl + + (Windows/Linux) ou Cmd + + (Mac), ou ajustar o tamanho da fonte diretamente em settings.json com "editor.fontSize": 16. Se quiser alterar só o chat sem mexer no editor, é necessário usar uma extensão de CSS customizado e definir o tamanho da fonte do painel do chat.

**Comment by Akash1134** [comment]
Hi @Iroegbu1 ,

GitHub Copilot X is currently a representation of GitHub’s vision for the future rather than an available product offering of GitHub Copilot. As we continue to design, test, and build features that fall into the GitHub Copilot X vision, we are also taking the time to determine the best way to provide them to our customers. 

Changing the size of the Chat window is not customizable at this point, I have shared this with the team internally. 

Thanks!

**Reply by Iroegbu1** [reply]
Hi @Akash1134 ,

> Changing the size of the Chat window

The **font** size...

The size of the chat window is currently resizeable.

**Reply by jnilla** [reply]
All of that double speech just to say the feature is not available. come on dude.

**Comment by pvsnp-hash** [comment]
Hi! If you go to settings and past: `chat.editor.fontSize ` in the seach bar you will find the setting that controls the font size.

**Reply by ahelmi365** [reply]
This increases the font size of the code snippet only, but it does not increase the font size of the other text in the chat response!

**Comment by ohnotnow** [comment]
Could this get looked at relatively soon?  I have terrible eyesight and am currently having to use the accessibility tools on my machine to read what copilot is writing.  The `chat.editor.fontSize` setting only seems to change the code-snippet font - which somehow makes it all the worse.

**Reply by Zephni** [reply]
I am having the same problem, the chat editor font size only changes the code snippet font size, not the entire chat text itself. I personally am really struggling to see it so I have to copy out the text into notepad to see the text 😂

![image](https://github.com/community/community/assets/4312475/5de71a68-7177-400e-9b02-73ab019b5c94)

**Reply by tit006** [reply]
Looks like the problem still exists.

**Reply by DancingLi** [reply]
It's 2025, and I'm having the same problem! Any solutions?

**Reply by anose001** [reply]
same here, can someone please help out with this

**Reply by marcosjesus99** [reply]
Yes, in 2025 mounth July problem I'm having the same problem, I wait any solution.

**Reply by siliconcoding** [reply]
> Yes, in 2025 mounth July problem I'm having the same problem, I wait any solution.

use cursor , use augment , ignore the Arrogant copilot ...

**Comment by ibehnam** [comment]
Is there any updates about this? The Copilot font size is painfully small.

**Comment by jackson-jackson** [comment]
Please fix asap.

**Comment by cloudbring** [comment]
Github Chat is a feature I use every day for work since it started. I get that it's experimental but, it's killing me, slowly, to have to look at TINY text.

**Comment by Lanznx** [comment]
this is so painful that i wrote a song, please, someone fix the damm feature

```
In the world of code, where ideas unfold,
There's a tool we use, it's bold yet old.
GitHub Copilot in VSCode, so bright,
But there's a little issue that's not quite right.

Oh, the font's too small, can't you see?
Straining our eyes, oh, set them free.
A simple fix, a wish from our heart,
Let's make Copilot Chat a more readable part.

Developers squint, in their coding chair,
Trying to read the text, it's just not fair.
Accessibility, a key we hold dear,
Adjustable fonts, let's bring them here.

Zooming in and out, it's not the way,
We need a better solution, day by day.
Eyes are tired, vision's getting blurry,
A simple font adjust, in a hurry!

Oh, the font's too small, in our coding spree,
A tweak in settings, how hard can it be?
Increase the size, for clarity's sake,
A comfortable read, for productivity's wake.

From light to dark, in every theme,
Let's make Copilot's text a readable dream.
For all who code, in day and night,
Adjustable fonts will make it right.

In the world of code, where we all strive,
Let's make our tools, help us thrive.
GitHub Copilot, hear our song,
Adjustable fonts, it won't be long!
```

wrote by ChatGPT

**Reply by allanpreis** [reply]
as

**Comment by Nik-at-Boolee** [comment]
Any updates on this one?

**Comment by aaronedev** [comment]
it cant be that hard to change the font-size of those windows - could we please get a fix? this is painful ☠

**Comment by ryandward** [comment]
Seems like it got worse recently

**Reply by opptimus** [reply]
I want to know when we can change the font size

**Reply by HamzaAxiom** [reply]
There is an old similar issue for font size change in vscode repo and it is been open since 2015!
https://github.com/microsoft/vscode/issues/519

**Reply by aaronedev** [reply]
lmao no way haha

**Comment by ntc2** [comment]
More potential workarounds in the related issue mentioned in [another comment](https://github.com/orgs/community/discussions/55469#discussioncomment-7981222): https://github.com/microsoft/vscode/issues/519#issuecomment-1687564021

**Comment by aaronedev** [comment]
> More potential workarounds in the related issue mentioned in [another comment](https://github.com/orgs/community/discussions/55469#discussioncomment-7981222): [microsoft/vscode#519 (comment)](https://github.com/microsoft/vscode/issues/519#issuecomment-1687564021)

thanks for sharing

**Comment by itseffi** [comment]
Any updates?

**Comment by sai3563** [comment]
Copilot Chat is now available generally so it's no longer "experimental".

Can we prioritize adding an option to increase font size now?

**Comment by qreeves** [comment]
Insert upvote here for immensely annoying lack of feature.

**Reply by opptimus** [reply]
I use pycharm ,I  want to vote this feature to new version of copilot plugin!

**Comment by kradyy** [comment]
I struggled with this myself and figured i'd shared a workaround i found that makes this possible without having to alter zoom levels and such - however, it _does_ requires you to install a plugin - if you don't mind doing that, here are my steps!

1. Head over to the extension panel and search for "Custom CSS and JS" - install it and reload vscode
2. Now, create a css file on your computer _(ie. editor.css)_ - preferably in a shared folder to be able to access it on different computers
3. Put the following contents into it (whereas 16px is your desired font size for the side panel):
`.pane-composite-part.right .content * {
  font-size: 16px !important;
}
`
3. Open your settings.json and add a [uri path](https://en.wikipedia.org/wiki/File_URI_scheme) for the file:
`"vscode_custom_css.imports": ["file:///Users/kradyy/Google%20Drive/vscode/editor.css"]`
4. Hit ⌘ + P (Mac) or CTRL + P (Windows) and look for "Enable Custom JS and CSS"
5. Restart vscode and voila - your font-size should now have been adjusted on your right sidebar

**Reply by LaeekAhmed** [reply]
thanks a lot, just a few things to note If it doesn't work right out of the box:

- Read the extension description
- Use `left` instead of `right` in the css code file if the chat panel is in the left side bar

**Reply by alexaldearroyo** [reply]
That's a very good workaround so far. Thanks for sharing!

**Comment by mo-ian-watkins** [comment]
Any news on this please?

**Comment by gbsr** [comment]
Updates?

**Comment by siliconcoding** [comment]
Updates?

**Comment by 0xrishavsharma** [comment]
Any update on this?

**Comment by szelonkaMS** [comment]
Please fix this.

**Comment by alexaldearroyo** [comment]
I'd like to join the request.

**Comment by mariushx** [comment]
Until they find time to fix this we have to go by manually editing the CSS file.
Locate workbench.desktop.main.css. On a default installation the file resides in C:\Users{user}\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench. Add these values at the end of the file and save.

```
:root {
  /* you can change these values */
  --vscode-copilot-chat-font-size: 16px; 
  --vscode-copilot-chat-font-family: Consolas, monospace;
  --vscode-copilot-chat-user-font-size: calc(var(--vscode-copilot-chat-font-size) - 3px);
}
.monaco-tl-contents .username {
  font-size: var(--vscode-copilot-chat-user-font-size) !important;
}
.monaco-button.interactive-followup-reply.monaco-text-button {
  font-size: var(--vscode-copilot-chat-font-size) !important;
}
.interactive-session .interactive-input-and-execute-toolbar .monaco-editor .mtk1,
.interactive-session .interactive-session-followups code {
  font-size: var(--vscode-copilot-chat-font-size) !important;
}
.interactive-item-container .monaco-tokenized-source, 
.interactive-item-container code{
    background-color: var(--vscode-textPreformat-background);
    border-radius: 4px;
    color: var(--vscode-textPreformat-foreground);
    font-family: var(--vscode-copilot-chat-font-family);
    font-size: var(--vscode-copilot-chat-font-size) !important;
    padding: 1px 3px;
}
.interactive-item-container .value > .rendered-markdown h3,
.interactive-item-container .value > .rendered-markdown p,
.interactive-item-container .value > .rendered-markdown li {
  font-size: var(--vscode-copilot-chat-font-size) !important;
  line-height: 1.5em !important;
}
```
Now if you are like me and you like to have font consistency across panels you should also set in settings.json file:
"window.zoomLevel": 0
"editor.suggestFontSize": 0
You should be set now.

**Reply by laurensiusadi** [reply]
This works. Just reload window.
VSCode will warn about corrupt installation.

**Reply by aaronedev** [reply]
thanks the best workaround probably

**Reply by consvs** [reply]
This is awesome! However, hopefully there will be specific setting to change the font size directly.

@laurensiusadi Did you manage to get rid of the warning? For now I just told vscode to never show this message again... 👎

**Reply by laurensiusadi** [reply]
@consvs Yeah same, just click never show.

**Reply by LeonardoRick** [reply]
@mariushx thanks for sharing. The path for mac users would be 

```
/Applications/Visual Studio Code App.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.css
```

Btw, do you have the styles to increase this part as well?

![image](https://github.com/user-attachments/assets/d9cf8600-6b27-48f9-aa7b-3d21d28eff8d)

**Reply by NicHub** [reply]
On my Mac the path is:
```bash
"/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.css"
```

**Reply by MrAmii** [reply]
This works but I was actually trying to change the font family so that everything matched up. When this didnt work I attempted to find and replace all instances of 

font-family:var(--monaco-monospace-font)
---->
font-family:` "Fira Code",  var(--monaco-mono-font)

Tried it this way because when I tried to pass it within, I don't know how because I don't know CSS like that. If you wouldn't mind guiding me, how would I pass Fira Code into your scripts font-family?

**Comment by MohammedNadeemR** [comment]
response 1

**Reply by mariushx** [reply]
If you don't mind having to zoom in the window, yes.

**Comment by tomx-sh** [comment]
Any updates ?
The chat font is too small, it is as small as UI elements (like buttons) - but the chat is _not_ UI, it is **actual content**!

**Comment by zoonose** [comment]
This is untested beyond what you see here - so beware of false hope - but I just asked copilot to write all its messages in code blocks (which are affected by chat.editor.fontSize, etc), and ...
```
// Understood. I will format all my messages like this from now on. 
// No code to execute here, just setting the format for future messages.
```
... :D !

**Reply by aaronedev** [reply]
well :D

**Comment by aaronedev** [comment]
i don't believe in microsoft ever delivering this update - i only believe in jesus christ 🧔🧔‍♂️✝️

**Comment by ajain-1** [comment]
There's a much simpler workaround. You can open chat in a new window (using the 3 dots in the top right of it), and then zoom in for just that window. This leaves the editor and everything else unaffected, it just zooms in for the chat window. You can then put the chat window alongside your editor as normal.

**Reply by LeonardoRick** [reply]
When I click "Open chat in sidebar" again, the increased zoom is lost and the font size returns to the default small font.

**Reply by ajain-1** [reply]
> When I click "Open chat in sidebar" again, the increased zoom is lost and the font size returns to the default small font. 

Yeah, don't open the chat in the sidebar, open it in a new window and you can zoom in/out independently

**Comment by movy** [comment]
I still have to use ChatGPT on web because of the minuscule font in sidebar chat. Are all Copilot devs on 85'' screens and this issue does not bother them at all?

**Comment by FindLeo** [comment]
I tried "ctrl +"
This will change all font but you can adjust the editor accordingly.

**Comment by henrychao24** [comment]
+1

**Comment by fk1blow** [comment]
Big AX issue on your part, harm on ours! 
Come on, just hack a button group with +/-  for now and call it a day, it doesn't have to be customizable through code's settings.

**Comment by mirumirumi** [comment]
Looks like software engineers worldwide have bad eyesight 😹

**Comment by 0xWeverton** [comment]
Any updates?

**Comment by derwaldgeist** [comment]
Seems as if they have no interest in us older users.

**Comment by movy** [comment]
I just got myself a pair of +0.75 spectacles (first pair in my life!), problem solved!  🤓

**Comment by scsskid** [comment]
+1

**Comment by 0xWeverton** [comment]
Temporary Solution: This guide helps you fix permission issues and enable custom CSS with the **Custom CSS and JS Loader** extension in VSCode Insiders on Linux.

## Steps

1. **Install the Extension**  
   - Open VSCode, go to the Extensions Marketplace (`Ctrl+Shift+X`), search for **"Custom CSS and JS Loader"**, and install it.

2. **Adjust Permissions**  
   Run these commands in your terminal:
   ```bash
   sudo chown -R $(whoami) /usr/share/code-insiders
   sudo chown -R $(whoami) $(which code-insiders)
   ```

3. **Enable Custom CSS**  
   - Restart VSCode.
   - Open the Command Palette (`Ctrl+Shift+P`), type **"Enable Custom CSS and JS"**, and select it.
   - If a "corrupted" warning appears, click **"Don't show again"**.
   - Restart VSCode again.

## Notes
- Ensure your `settings.json` has the CSS path (e.g., `"vscode_custom_css.imports": ["file:///home/your-username/.vscode-custom/custom.css"]`).
- Need create custom.css with code bellow

---

custom.css:
```css

.pane-composite-part.right .content * { 
    font-size: 19px !important; 
}

```

**Reply by NicHub** [reply]
Good idea. It is also possible to inject the CSS from the developer tools console :

```
Developer: Toggle Developer Tools (Ctrl+Shift+P)
```

click on `Console` on the top left.

First time only: type `allow pasting`

Then copy-paste this snippet:

```js
const style = document.createElement("style");
document.head.appendChild(style);
style.textContent = ".pane-composite-part.right .content * { font-size: 19px !important; }";
```

If you want to try other sizes, execute the last line only.

The advantage of this solution is that you don’t have to install an extension.

The disadvantage is that you have to redo it each time you reopen VSCode and for each workspace, but note that VSCode console keeps the history so you can recall the snippet by pressing Arrow up in the console.

**Comment by aaronedev** [comment]
nevermind i switched to neovim

**Comment by sudg6** [comment]
for me, this rule works as expected:
```
.monaco-workbench .part>.content {
    font-size: 16px;
}
```
but not sure how to make it persistent without **Open the Command Palette (Ctrl+Shift+P), type "Enable Custom CSS and JS", and select it.** each time I launch `code-insiders`

**Comment by martysai** [comment]
selling the garage.

**Comment by Bravi-study** [comment]
Is there any hope left?

**Comment by LeninZapata** [comment]
May 2025, any updates? The copilot's replies have a larger font than the chat font.

**Comment by kodewulf** [comment]
It's stupidly ridiculous that this is still an issue after 2 YEARS!  VSCODE accessibility features are almost non-
existent. This is what happens when trends and profit drives open source development.

**Comment by derwaldgeist** [comment]
I'm using the chat in WebStorm. The font size is incredibly small. Almost unusable.

Besides, I noticed that the quality is much worse than ChatGPT. Are they going to do something about that as well, or is there a way to switch to a better model? Quite often, I am running in circles with the Chatbot.

**Comment by mykolav** [comment]
Here's a semi-permanent workaround.
Append the following CSS code to the end of `$VSCODE_INSTALL_DIR/resources/app/out/vs/workbench/workbench.desktop.main.css`.

```
 /* Increase font size in the Copilot Chat panel */
div#workbench\.panel\.chat {
  font-size: 14px !important;
}

/* Also target descendant text elements */
div#workbench\.panel\.chat * {
  font-size: 14px !important;
}
```

Saving the file will cause a warning to show up, saying that the installation appears to be corrupt. I personally just silenced it.

The font in the chat window (and only there) will be larger after restarting VS Code.

Keep in mind that a VS Code update will likely overwrite this file.

**Reply by LeninZapata** [reply]
Okay, it worked, but I have a question about the line-height. When I apply it, it only works on the chat text, but not on the code box. Can the line-height be adjusted for the code? 
```
div#workbench\.panel\.chat {
  font-size: 12px !important;
  line-height: 15px !important;
}

/* Also target descendant text elements */
div#workbench\.panel\.chat * {
  font-size: 12px !important;
  line-height: 15px !important;
}
```

**Comment by BerndHme** [comment]
I just found out you can open copilot chat in a separate window.
You can adjust the editor (separate window) fontsize separatly.
Also, you can set the new window "always on top".

![image](https://github.com/user-attachments/assets/53c3e449-d157-4161-8577-cb5eb4987707)

**Reply by NicHub** [reply]
Best answer so far.

**Reply by L-N1988** [reply]
you are my hero, cool solution!!!

**Reply by tylovehhx** [reply]
brilliant！

**Reply by derwaldgeist** [reply]
> I just found out you can open copilot chat in a separate window.

Unfortunately, this does not work in the JetBrains IDEs.

**Comment by anuvindvasp** [comment]
Open VS Code.

Press Ctrl + , (or go to File > Preferences > Settings).

In the search bar, type: editor.fontSize


Or try Workbench > Font Size

Also Press Ctrl + + to zoom in on the interface

**Reply by aaronedev** [reply]
> Open VS Code.
> 
> Press Ctrl + , (or go to File > Preferences > Settings).
> 
> In the search bar, type: editor.fontSize
> 
> Or try Workbench > Font Size
> 
> Also Press Ctrl + + to zoom in on the interface

bro for real? 😂😂😂

**Comment by fk1blow** [comment]
First, a genuine question: Why not use Copilot agent mode to come up with a solution for this? It seems like exactly the kind of problem AI could help solve efficiently.

Second, appreciation: We're genuinely grateful for VSCode and the incredible work the team puts into this project. Most of us truly appreciate what you've built and continue to build.

However, the accessibility concern: Why is this considered such a low priority task when it's clearly an accessibility issue—and a pretty significant one at that?

This isn't just about preference—it's about making the tool accessible to users with different visual needs. And let's be honest, this affects most developers, not just those with existing visual impairments. As we spend more hours coding and age, eye strain and vision changes become increasingly common. Many of us already find ourselves needing different font sizes for different types of content.

The fundamental issue is that these are two completely different interfaces serving entirely different purposes. Code requires one type of visual processing (syntax highlighting, precise character recognition, line-by-line reading), while chat conversations require another (natural language flow, paragraph scanning, dialogue comprehension). Treating them as a single entity makes no ergonomic sense.

The current coupling of chat interface font settings with the entire editor creates real usability barriers for anyone who needs different font configurations for reading conversational text versus code.

We cannot understand why this accessibility concern continues to be disregarded, especially when it's been raised by multiple users across various threads.

Would love to see this addressed in a future update. Thanks for considering our feedback!

**Reply by gh-andre** [reply]
Very well put. I wish the Product Owner for this area would see the distinction between a work-around with the new window and <kbd>Ctrl-+</kbd> and a dedicated setting to configure how this very important feature is presented in the UI _every time_ the Copilot window is opened, docked or not.

On the <kbd>Ctrl-+</kbd> front, I didn't notice it because every single UI I ever worked with, including Visual Studio, zooms with <kbd>Ctrl-mouse-wheel</kbd>, well, except VSCode.

**Reply by derwaldgeist** [reply]
I fully agree with this. And many of us aren't in their twenties anymore. At that time, I could read literally any size. Not anymore.

**Comment by rahbari** [comment]
the best data that can't be read is useless.

**Comment by Debangi2004** [comment]
Zoom In (No Extensions Needed) by pressing Ctrl & + or Click the 3 dots ⋮ in the top-right corner ->Go to Settings > Appearance ->Adjust Font size to "Large" or "Very Large"

**Comment by ependergrass** [comment]
My problem is when I use Eclipse w/ADT (ABAP development tools) in dark mode, I can't see portions of the header text in Github Copilot Chat. Does anyone else have this issue??

**Comment by 0xWeverton** [comment]
How can the team not see the amount of feedback we need for something as simple as just being able to change the font size of the active window? It should be something simple, and we’ve been asking for more than 2 years with no response. What is going on?

**Reply by derwaldgeist** [reply]
Seems as if they don't want users.

**Comment by gh-andre** [comment]
I suppose assigning issues to Copilot and having it write code on its own is more of a marketing thing, then.

**Comment by Andonvr** [comment]
This is a real strain on my eyes, wish this got solved :(

**Comment by russeg** [comment]
Increasing the font size on the side bar is a very very hard problem!  This requires some super complex computer science solution. I don't think the VS Code team can handle this 😆

**Reply by derwaldgeist** [reply]
Maybe they could use AI to solve this? 🧐

**Comment by drgrib** [comment]
These are the golden selectors from what I have discovered so far. Put these in your User level `settings.json`. Change the font sizes as desired (note the last one needs to be a smaller size than the others to look right). Approve the UI changes and let it restart the app after saving:

``` json
	"custom-ui-style.stylesheet": {
		".rendered-markdown": "font-size: 17px !important;",
		".interactive-response .monaco-editor .view-lines": "font-size: 17px !important;",
		".chat-markdown-part code": "font-size: 17px !important;",
		".interactive-input-part .monaco-editor .view-lines": "font-size: 17px !important;",
		".chat-tool-invocation-part .progress-step p": "font-size: 17px !important;",
		".interactive-session .chat-used-context-label": "font-size: 17px !important;",
		".interactive-session .chat-used-context-label code": "font-size: 14px !important;"
	}
```

For an even deeper cut, which will increase the size of the context buttons and files changed at the bottom, you can use this. Just note that there are some more smaller sizes at the end and **_this will change your sidebar and tab text sizes_**. I like how it looks but your mileage may vary:

``` json
	"custom-ui-style.stylesheet": {
		".rendered-markdown": "font-size: 17px !important;",
		".interactive-response .monaco-editor .view-lines": "font-size: 17px !important;",
		".chat-markdown-part code": "font-size: 17px !important;",
		".interactive-input-part .monaco-editor .view-lines": "font-size: 17px !important;",
		".chat-tool-invocation-part .progress-step p": "font-size: 17px !important;",
		".interactive-session .chat-used-context-label": "font-size: 17px !important;",
		".interactive-session .chat-used-context-label code": "font-size: 14px !important;",
		".action-item.chat-attachment-button .action-label, .interactive-session .chat-attached-context .chat-attached-context-attachment": "font-size: 15px !important;",
		".monaco-icon-label .label-name": "font-size: 15px !important;",
		".interactive-session .chat-editing-session .chat-editing-session-container .chat-editing-session-overview > .working-set-title": "font-size: 15px !important;",
		".monaco-icon-label .label-description": "font-size: 15px !important;"
	}
```

### Updating the Rules
To find more or adjust for updates, open up your favorite LLM (I use Gemini 2.5 Pro), give it the above json and tell it you want to update it. Then activate `Developer: Toggle Developer Tools`. From there, 2 approaches have worked for me, in order of ease and effectiveness.
#### Method 1
1. Search the HTML for your text
2. Try to select and isolate the closest tag but most you need resized
3. Copy the CSS pane contents for that tag
4. Give that to the LLM and ask it to update the JSON.
#### Method 2
1. edit top tag as HTML
2. copy to file
3. search for desired element text to change
4. copy as much HTML context as possible around your target tag and give it to the LLM to update the JSON (you likely won't be able to give it the entire file because it will be too big)

**Comment by Kunalbairwa07** [comment]
Hi! You can increase the chat font size from your browser’s zoom settings (e.g., Ctrl + + on Windows or Cmd + + on Mac). GitHub doesn’t currently have a built-in font size setting for Discussions.

**Reply by drgrib** [reply]
The problem with that solution is that it also changes the size of everything else and looks really janky, imo. My solution (posted directly above) allows you to isolate the changes.

**Comment by dslijepcevic** [comment]
To all brothers and sisters, fellow squinters… it's finally done! 🎉

The latest VSCode 1.104 (August 2025) brings the long-awaited chat font settings —
`chat.fontFamily` and `chat.fontSize`!

🙏

**Reply by derwaldgeist** [reply]
What about us JetBrains users?

**Reply by dslijepcevic** [reply]
Umm, scratch that, the input field font is still too small, not sure how to change that one.

**Reply by GiannisTar** [reply]
the goat

**Comment by LeninZapata** [comment]
for chat text: `chat.fontSize`
for chat code: `chat.editor.fontSize`

**Reply by dslijepcevic** [reply]
No, that only affects font size within the chat itself. The input field at the bottom of the chat where you write prompts is still super tiny.

**Reply by stephanemagnenat** [reply]
Yes, this is really annoying, please either use the same font size as chat text for the input field, or add an additional one.

**Comment by vickyshare66666** [comment]
I can find that in settings at the left bottom corner in 2026.
Search " font size" , then list many relevant font size setting.
Find the settings below:
Chat: Font Size
Controls the font size in pixels in chat messages.

**Reply by Floyddotnet** [reply]
settings.json: `"chat.fontSize": 15`

**Comment by demcy** [comment]
What about Rider IDE?

**Comment by mistervaibhav** [comment]
mother of god, how is this still an issue 3 years later

**Comment by garv999** [comment]
As of recent VS Code versions, this issue is finally addressed.

If you're using VS Code, open **Settings** (`Ctrl + ,`) and search for **Chat: Font Size**, or add the following to `settings.json`:

```json
"chat.fontSize": 15
```

You can also adjust code blocks inside chat separately with:

```json
"chat.editor.fontSize": 15
```

Note that the prompt input box may still use a smaller font and currently doesn't have its own setting.

For JetBrains IDEs (Rider, WebStorm, etc.), there is still no dedicated Copilot Chat font size setting available.
